dali/ft/filecopy.cpp

```diff
@@ -3224,8 +3224,68 @@ void FileSprayer::updateTargetProperties()
             }
 
             // Keep source kind
-            curProps.setProp("@kind", srcAttr->queryProp("@kind"));
+            if (srcAttr->hasProp("@kind"))
+                curProps.setProp("@kind", srcAttr->queryProp("@kind"));
+            else
+            {
+                StringBuffer targetKind = "csv";
+
+                if (srcFormat.markup == FMTxml)
+                    targetKind = "xml";
+                else if (srcFormat.markup == FMTjson)
+                    targetKind = "json";
+
+                StringBuffer targetFormat = "utf8n";
+
+                switch (srcFormat.type)
```

Comment:
Is this right? 
Shouldn't tgtFormat.type be used here?

The source format is not necessarily going to be same as target format.

Reply:
Changed